args = commandArgs(trailingOnly = TRUE)

library(numbat)

count_mat_ATC2 = readRDS(args[1])

df_allele_ATC2 = readRDS(args[2])

out_dir = args[3]
ncores = 2

# run
out = run_numbat(
  count_mat_ATC2, # gene x cell integer UMI count matrix 
  ref_hca, # reference expression profile, a gene x cell type normalized expression level matrix
  df_allele_ATC2, # allele dataframe generated by pileup_and_phase script
  genome = "hg38",
  t = 1e-5,
  ncores = ncores,
  plot = TRUE,
  out_dir = out_dir # '../../../test_run_numbat'
)

